dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/MetadataDumper.java

```diff
@@ -229,21 +228,11 @@ protected void run(@Nonnull Connector connector, @Nonnull ConnectorArguments arg
         // We had a customer request to base it on the database, but that isn't well-defined,
         // as there may be 0 or N databases in a single file.
         File outputFile = arguments.getOutputFile();
-        String outputDirectory = arguments.getOutputDirectory();
-
-        boolean isDefaultPath = "".equals(outputDirectory);
-        if (!isDefaultPath && outputFile != null) {
-            System.out.println(
-                "**********************************************************\n"
-                + "* ERROR: Using both --output and --output-dir flags is not allowed.\n"
-                + "* Please use --help for more information.\n"
-                + "**********************************************************"
-            );
-            return;
-        }
-
-        if (outputFile == null) {
-            outputFile = new File(outputDirectory, connector.getDefaultFileName(arguments.isAssessment()));
+        String defaultFileName = connector.getDefaultFileName(arguments.isAssessment());
+        if (outputFile == null || outputFile.getName().isEmpty()) {
```

Comment:
I would suggest to avoid null, wherever possible. In this case, I would recommend changing the return type of `ConnectorArguments#getOutputFile()` method to `Optional<File>`. In addition to `null`, empty string can also be represented as empty optional, so that this condition is simplified.

See example:
https://github.com/google/dwh-migration-tools/blob/3161155bdbb796c30f8caee89d15276ad2032f74/dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/ConnectorArguments.java#L441